src/GitHub.App/Services/NavigationService.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.TextManager.Interop;
+using IServiceProvider = System.IServiceProvider;
+using GitHub.Models;
+
+namespace GitHub.Services
+{
+    [Export(typeof(INavigationService))]
+    public class NavigationService : INavigationService
+    {
+        readonly IServiceProvider serviceProvider;
+
+        [ImportingConstructor]
+        public NavigationService([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider)
+        {
+            this.serviceProvider = serviceProvider;
+        }
+
+        public IVsTextView NavigateToEquivalentPosition(IVsTextView sourceView, string targetFile)
+        {
+            int line;
+            int column;
+            ErrorHandler.ThrowOnFailure(sourceView.GetCaretPos(out line, out column));
```

Comment:
If this fails, what will catch this throw?

Reply:
If this fails it will give hint as to what went wrong on the status bar: `ShowErrorInStatusBar("Error opening live file", e);`. I'm handling errors in a similar way to other methods in `PullRequestDetailView.xaml.cs`.